src/Riok.Mapperly/Descriptors/WellKnownTypes.cs

```diff
@@ -1,127 +1,54 @@
-using System.Collections.Immutable;
 using Microsoft.CodeAnalysis;
-using Riok.Mapperly.Abstractions;
-using Riok.Mapperly.Abstractions.ReferenceHandling;
-using Riok.Mapperly.Abstractions.ReferenceHandling.Internal;
 
 namespace Riok.Mapperly.Descriptors;
 
 public class WellKnownTypes
 {
     private readonly Compilation _compilation;
+    private readonly Dictionary<string, INamedTypeSymbol> _cachedTypes = new();
 
-    private INamedTypeSymbol? _referenceHandlerAttribute;
-    private INamedTypeSymbol? _objectFactoryAttribute;
-    private INamedTypeSymbol? _mapperConstructorAttribute;
-
-    private INamedTypeSymbol? _obsoleteAttribute;
+    internal WellKnownTypes(Compilation compilation)
+    {
+        _compilation = compilation;
+    }
 
-    private INamedTypeSymbol? _iReferenceHandler;
-    private INamedTypeSymbol? _preserveReferenceHandler;
+    public INamedTypeSymbol GetTypeSymbol<T>() => GetTypeSymbol(typeof(T).FullName);
 
-    private INamedTypeSymbol? _iDictionaryT;
-    private INamedTypeSymbol? _iReadOnlyDictionaryT;
-    private INamedTypeSymbol? _iEnumerableT;
-    private INamedTypeSymbol? _enumerable;
-    private INamedTypeSymbol? _iCollectionT;
-    private INamedTypeSymbol? _iReadOnlyCollectionT;
-    private INamedTypeSymbol? _iListT;
-    private INamedTypeSymbol? _listT;
-    private INamedTypeSymbol? _stackT;
-    private INamedTypeSymbol? _queueT;
-    private INamedTypeSymbol? _iReadOnlyListT;
-    private INamedTypeSymbol? _keyValuePairT;
-    private INamedTypeSymbol? _dictionaryT;
-    private INamedTypeSymbol? _enum;
-    private INamedTypeSymbol? _type;
+    public INamedTypeSymbol GetTypeSymbol(Type type) =>
```

Comment:
I think these method names could be simplified to `Get`/`TryGet` as I think it should be clear that it is about types as the class is named `WellKnownTypes`.